notebooks/04_model_select_and_optimize/nni_surprise_svd.ipynb

```diff
@@ -0,0 +1,1066 @@
+   "execution_count": 38,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# SMAC\n",
+    "config['tuner']['builtinTunerName'] = 'SMAC'\n",
+    "with open(config_path, 'w') as fp:\n",
+    "    fp.write(yaml.dump(config, default_flow_style=False))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Check if installed\n",
+    "proc = subprocess.run([sys.prefix + '/bin/nnictl', 'package', 'show'], stdout=subprocess.PIPE)\n",
+    "if proc.returncode != 0:\n",
+    "    raise RuntimeError(\"'nnictl package show' failed with code %d\" % proc.returncode)\n",
+    "if 'SMAC' not in proc.stdout.decode().strip().split():\n",
+    "    proc = subprocess.run([sys.prefix + '/bin/nnictl', 'package', 'install', '--name=SMAC'])\n",
+    "    if proc.returncode != 0:\n",
+    "        raise RuntimeError(\"'nnictl package install' failed with code %d\" % proc.returncode)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 39,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "t = time.time()\n",
+    "stop_and_restart()\n",
+    "time_smac = time.time() - t"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 40,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "check_metrics_written()\n",
+    "svd = surprise.dump.load(os.path.join(get_trials('maximize')[3], \"model.dump\"))[1]\n",
+    "test_results_smac = compute_test_results(svd)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 41,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Metis\n",
+    "config['tuner']['builtinTunerName'] = 'MetisTuner'\n",
+    "with open(config_path, 'w') as fp:\n",
+    "    fp.write(yaml.dump(config, default_flow_style=False))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 43,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "t = time.time()\n",
+    "stop_and_restart()\n",
+    "time_metis = time.time() - t"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 44,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "check_metrics_written()\n",
+    "svd = surprise.dump.load(os.path.join(get_trials('maximize')[3], \"model.dump\"))[1]\n",
+    "test_results_metis = compute_test_results(svd)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Hyperband follows a different style of configuration from other tuners. See [the NNI documentation](https://nni.readthedocs.io/en/latest/hyperbandAdvisor.html). Note that the [training script](../../reco_utils/nni/svd_training.py) needs to be adjusted as well, since each Hyperband trial receives an additional parameter `STEPS`, which corresponds to the resource allocation _r<sub>i</sub>_ in the [Hyperband algorithm](https://arxiv.org/pdf/1603.06560.pdf). In this example, we used `STEPS` in combination with `R` to determine the number of epochs that SVD will run for in every trial."
```

Comment:
We will change `STEPS` to `TRIAL_BUDGET` for easy understanding in the next release v0.7, very sorry for this broken change. We will inform you when v0.7 is released. Thanks.

Reply:
Thanks for the notice

Reply:
Thanks for the notice